File: javascript/word-count/word-count.js
function Words() {
  this.count = function (words) {
    var wordCount = {};
    words = words.trim();
    words.split(/[ \n\t]+/).forEach(function (word) {
      word = word.toLowerCase();
      wordCount[word] = (+wordCount[word] || 0) + 1;
    });
    return wordCount;
  }
}

module.exports = Words;

File: python/word-count/wordcount.py
import re


def word_count(sentence):
    words = re.split("(?u)\W+|_", sentence.decode("utf-8").lower())
    words = filter(None, words)
    count = {}
    for word in words:
        count[word] = 1 if word not in count else count[word] + 1
    return count
